src/query/plan/operator.cpp

```diff
@@ -5999,21 +6004,48 @@ class PeriodicCommitCursor : public Cursor {
   }
 
   bool Pull(Frame &frame, ExecutionContext &context) override {
+    // NOLINTNEXTLINE(misc-const-correctness)
     OOMExceptionEnabler oom_exception;
     const SCOPED_PROFILE_OP_BY_REF(self_);
 
-    throw utils::NotYetImplemented("periodic commit");
+    if (commit_frequency_ == -1) {
+      ExpressionEvaluator evaluator(&frame, context.symbol_table, context.evaluation_context, context.db_accessor,
+                                    storage::View::OLD);
+      TypedValue commit_frequency = self_.commit_frequency_->Accept(evaluator);
+      if (commit_frequency.type() != TypedValue::Type::Int)
+        throw QueryRuntimeException("Limit on number of returned elements must be an integer.");
```

Comment:
I don't think that error message is correct for this case, can we change it?

Reply:
abstracted this into a function as per this comment and the next one